packages/client/tests/functional/issues/13913-integer-overflow/tests.ts

```diff
@@ -0,0 +1,27 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(() => {
+  test('int overflow', async () => {
+    const promise = prisma.resource.create({
+      data: {
+        number: 2265000000,
+      },
+    })
+
+    await expect(promise).rejects.toMatchPrismaErrorInlineSnapshot(`
+
+      Invalid \`prisma.resource.create()\` invocation in
+      /client/tests/functional/issues/13913-integer-overflow/tests.ts:0:0
+
+        XX 
+        XX testMatrix.setupTestSuite(() => {
+        XX   test('int overflow', async () => {
+      → XX     const promise = prisma.resource.create(
+      Failed to validate the query: \`Unable to match input value to any allowed input type for the field. Parse errors: [Query parsing/validation error at \`Mutation.createOneResource.data.ResourceCreateInput.number\`: Unable to fit integer value '2265000000' into a 32-bit signed integer for field 'number'., Query parsing/validation error at \`Mutation.createOneResource.data.ResourceUncheckedCreateInput.number\`: Unable to fit integer value '2265000000' into a 32-bit signed integer for field 'number'.]\` at \`Mutation.createOneResource.data\`
```

Comment:
This looks like a "bad", unhandled error message that should get a proper error code and name if we keep doing this.

Reply:
The error looks ok, it also has a `P2023` error code. The message isn't super user-friendly though `Mutation.createOneResource.data.ResourceUncheckedCreateInput.number`, this is an implementation detail probably leaking into the message.

Reply:
Are we just skipping the error code part in our snapshots?

Reply:
Yes, I assume this is because the `code` property is not enumerable. We should fix that.

Reply:
:+1: Def confused me.